File: 백준/Silver/29700. 우당탕탕 영화예매/우당탕탕 영화예매.py
import sys
input=sys.stdin.readline

n,m,k=map(int,input().split())
data=[]
answer=0

for _ in range(n):
  #가로줄 입력 받기
  row=input().rstrip()
  for i in range(m-k+1):
    # i부터 (i+k-1)까지 모두 예매 가능한 경우
    if '1' not in row[i:i+k]:
      answer+=1
      
print(answer)

File: 백준/Silver/29700. 우당탕탕 영화예매/우당탕탕 영화예매.java
import java.io.*;
import java.util.*;

public class Main {
    
  public static void main(String args[]) throws IOException {
      BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
      StringTokenizer st = new StringTokenizer(br.readLine());
      
      int N = Integer.parseInt(st.nextToken());
      int M = Integer.parseInt(st.nextToken());
      int K = Integer.parseInt(st.nextToken());
      
      int answer = 0;
      
      for(int i=0; i<N; i++){
          String row = br.readLine();
          String[] segments = row.split("1");
          
          for(String seg: segments){
              if(seg.length()>=K){
                  answer+=seg.length()-K+1;
              }
          }
      }
      
      System.out.println(answer);
      
      br.close();
  }
}